doc/nrf/libraries/modem/nrf_modem_lib.rst

```diff
@@ -101,6 +101,82 @@ If the modem trace medium is unable to keep up with the modem traces, the heap s
 Increasing the heap size allows more traces in the FIFO queue, but the trace heap will still be depleted if the modem continues to send traces at a rate faster than the rate at which the medium can handle over time.
 If increasing the trace heap size does not help, either optimize the medium speed or use a faster trace transport medium.
 
+.. _adding_custom_modem_trace_transport_mediums:
+
+Adding custom trace transport mediums
+=====================================
+
+If the existing trace transport mediums are not sufficient, custom transport mediums can be added.
+Note that only one transport medium can be compiled with the application at any time.
+Which trace medium that is compiled is controlled by the Kconfig choice :kconfig:option:`NRF_MODEM_LIB_TRACE_MEDIUM`.
+
+Adding a custom transport medium can be done by following the steps below.
+Place the custom transport medium implementation in a library or an application you create.
+As a reference, the implementation of the UART trace medium (default medium) can be found in :file:`nrf/lib/nrf_modem_lib/trace_mediums`.
+The :ref:`custom_trace_medium_sample` sample demonstrates how a custom trace medium can be added to an application.
+
+1. Add a C file implementing the interface in :file:`nrf/include/modem/trace_medium.h`\.
+   .. code-block:: c
+   /* my_trace_medium.c */
+   #include <modem/trace_medium.h>
+
+   int trace_medium_init(void)
+   {
+           /* initialize transport medium here */
+           return 0;
+   }
+
+   int trace_medium_deinit(void)
+   {
+           /* optional deinitialization code here */
+           return 0;
+   }
+
+   int trace_medium_write(const void *data, size_t len)
+   {
+           /* forward data over custom transport here */
+           /* return number of bytes written or negative error code on failure */
+   }
+
+#. Add or modify a :file:`Kconfig` file, extending the choice :kconfig:option:`NRF_MODEM_LIB_TRACE_MEDIUM` with another option.
+   .. code-block:: Kconfig
+      if NRF_MODEM_LIB_TRACE_ENABLED
+
+      # Extends the choice with another medium
+      choice NRF_MODEM_LIB_TRACE_MEDIUM
+
+      config NRF_MODEM_LIB_TRACE_MEDIUM_MY_TRACE_MEDIUM
+              bool "My trace transport medium"
+              help
+                Optional description of my
+                trace transport medium.
+
+      endchoice
+
+      endif
```

Comment:
```suggestion
#. Create or modify a :file:`Kconfig` file to set the :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM` Kconfig option with another option.

   .. code-block:: Kconfig
   
      if NRF_MODEM_LIB_TRACE_ENABLED
      
      # Extends the choice with another medium
      choice NRF_MODEM_LIB_TRACE_MEDIUM  

      config NRF_MODEM_LIB_TRACE_MEDIUM_MY_TRACE_MEDIUM
             bool "My trace transport medium"
             help
              Optional description of my
              trace transport medium.

      endchoice

      endif
```

Reply:
@divipillai This particular Kconfig code is not setting the option but extending it